packages/frontend/pages/strategies.tsx

```diff
@@ -115,44 +116,48 @@ const useStyles = makeStyles((theme) =>
     },
     toggle: {
       justifyContent: 'right',
-      marginLeft: theme.spacing(2)
+      marginLeft: theme.spacing(2),
     },
   }),
 )
 
 const Strategies: React.FC = () => {
   const [selectedIdx, setSelectedIdx] = useState(1)
 
-   // which crab strategy to display. V1 or V2. 
+  // which crab strategy to display. V1 or V2.
   const [displayCrabV1, setDisplayCrabV1] = useState(false)
 
   const classes = useStyles()
-  const maxCap = displayCrabV1? useAtomValue(maxCapAtom) : useAtomValue(maxCapAtomV2)
-  const vault = displayCrabV1? useAtomValue(crabStrategyVaultAtom) : useAtomValue(crabStrategyVaultAtomV2) 
-  const collatRatio = displayCrabV1? useAtomValue(crabStrategyCollatRatioAtom) : useAtomValue(crabStrategyCollatRatioAtomV2)
-  const timeAtLastHedge = displayCrabV1? useAtomValue(timeAtLastHedgeAtom) : useAtomValue(timeAtLastHedgeAtomV2)
+  const maxCap = useAtomValue(displayCrabV1 ? maxCapAtom : maxCapAtomV2)
+  const vault = useAtomValue(displayCrabV1 ? crabStrategyVaultAtom : crabStrategyVaultAtomV2)
+  const collatRatio = useAtomValue(displayCrabV1 ? crabStrategyCollatRatioAtom : crabStrategyCollatRatioAtomV2)
+  const timeAtLastHedge = useAtomValue(displayCrabV1 ? timeAtLastHedgeAtom : timeAtLastHedgeAtomV2)
   const profitableMovePercent = useSetProfitableMovePercent()
-  const setStrategyData = displayCrabV1? useSetStrategyData() : useSetStrategyDataV2()
+  const setStrategyData = useSetStrategyData()
+  const setStrategyDataV2 = useSetStrategyDataV2()
+
   useCurrentCrabPositionValueV2()
   useCurrentCrabPositionValue()
   useInitCrabMigration()
 
   const index = useAtomValue(indexAtom)
-  const isQueued = useAtomValue(isQueuedAtom)
   const dailyHistoricalFunding = useAtomValue(dailyHistoricalFundingAtom)
   const currentImpliedFunding = useAtomValue(currentImpliedFundingAtom)
-  const currentEthValue = displayCrabV1? useAtomValue(currentCrabPositionValueInETHAtom) : useAtomValue(currentCrabPositionValueInETHAtomV2)
 
   const address = useAtomValue(addressAtom)
   const supportedNetwork = useAtomValue(supportedNetworkAtom)
   const selectWallet = useSelectWallet()
 
-  const CapDetailsComponent = displayCrabV1? CapDetails : CapDetailsV2
-  const CrabTradeComponent = displayCrabV1? CrabTrade: CrabTradeV2
+  const CapDetailsComponent = displayCrabV1 ? CapDetails : CapDetailsV2
+  const CrabTradeComponent = displayCrabV1 ? CrabTrade : CrabTradeV2
+
+  useEffect(() => {
+    if (displayCrabV1) setStrategyData()
+  }, [collatRatio, displayCrabV1, setStrategyData])
 
   useEffect(() => {
-    setStrategyData()
-  }, [collatRatio, setStrategyData])
+    if (!displayCrabV1) setStrategyDataV2()
```

Comment:
Is the PnL right for V1? For V2, I had to call `setStrategyDataV2()` to have it update properly. But it looks reasonable for V2 here.
<img width="848" alt="Screen Shot 2022-07-25 at 10 30 21 AM" src="https://user-images.githubusercontent.com/29926727/180841170-3818cc88-8438-4916-a480-e247da20bed1.png">

Reply:
Based on the displayCrabV1 it either calls setStrategyDataV2 or setStrategyData. So should not be a problem